src/app/features/activity-list/components/transactions/transaction-list-item.tsx

```diff
@@ -0,0 +1,26 @@
+import { BitcoinTransactionItem } from '@app/components/transaction/bitcoin/bitcoin-transaction-item';
+import { WalletBlockchains } from '@shared/models/blockchain.models';
+import { AllWalletTransactions } from '@shared/models/transactions/transactions.models';
+import { StacksTransaction } from './stacks-transaction';
+
+function renderTransaction(tx: AllWalletTransactions) {
+  switch (tx.blockchain) {
+    case WalletBlockchains.bitcoin:
+      return <BitcoinTransactionItem transaction={tx.transaction} />;
+    case WalletBlockchains.stacks:
+      return (
+        <StacksTransaction
+          transaction={tx.transactionWithTransfers || tx.mempoolTransaction || tx.transaction}
```

Comment:
Not 100% on the type being one of three options. Perhaps it needs to be this way, but would love to dig deeper and understand why we need both `transactionWithTransfers` and `transaction`. Maybe we can talk about this next week

Reply:
Yeah, maybe I can remove `transaction` here ...I believe it was being pulled off `transactionWithTransfers` during the grouping so maybe I added by mistake. I'll remove if can which is likely.